src/bindgen/ir/ty.rs

```diff
@@ -270,6 +270,14 @@ impl ConstExpr {
                 let generic_path = GenericPath::load(&path.path)?;
                 Ok(ConstExpr::Name(generic_path.export_name().to_owned()))
             }
+            syn::Expr::Cast(ref cast) => {
```

Comment:
Just `return ConstExpr::load(&cast.expr)`, that way it also deals with `4 as usize` or what not.